adapters/somoaudience/somoaudience.go

```diff
@@ -4,81 +4,158 @@ import (
 	"encoding/json"
 	"fmt"
 	"net/http"
+	"strconv"
 
+	"github.com/golang/glog"
 	"github.com/mxmCherry/openrtb"
 	"github.com/prebid/prebid-server/adapters"
 	"github.com/prebid/prebid-server/errortypes"
 	"github.com/prebid/prebid-server/openrtb_ext"
 )
 
+const config = "hb_pbs_1.0.0"
+
 type SomoaudienceAdapter struct {
 	endpoint string
 }
 
+type somoaudienceReqExt struct {
+	BidderConfig string `json:"prebid"`
+}
+
 func (a *SomoaudienceAdapter) MakeRequests(request *openrtb.BidRequest) ([]*adapters.RequestData, []error) {
 
-	totalImps := len(request.Imp)
-	errors := make([]error, 0, totalImps)
-	imp2placement := make(map[string][]int)
+	var errs []error
+	var bannerImps []openrtb.Imp
+	var videoImps []openrtb.Imp
+	var nativeImps []openrtb.Imp
+
+	for _, imp := range request.Imp {
+		// OpenX doesn't allow multi-type imp. Banner takes priority over video.
+		if imp.Banner != nil {
+			bannerImps = append(bannerImps, imp)
+		} else if imp.Video != nil {
+			videoImps = append(videoImps, imp)
+		} else if imp.Native != nil {
+			nativeImps = append(nativeImps, imp)
+		} else {
+			err := &errortypes.BadInput{
+				Message: fmt.Sprintf("SomoAudience only supports banner and video imps. Ignoring imp id=%s", imp.ID),
+			}
+			glog.Warning("SomoAudience CAPABILITY VIOLATION: only supports banner and video imps")
+			errs = append(errs, err)
+		}
+	}
+	var adapterRequests []*adapters.RequestData
+	// Make a copy as we don't want to change the original request
+	reqCopy := *request
+
+	reqCopy.Imp = bannerImps
+	adapterReq, errors := a.makeRequest(&reqCopy)
+	if adapterReq != nil {
+		adapterRequests = append(adapterRequests, adapterReq)
+	}
+	errs = append(errs, errors...)
+
+	// Somoaudience only supports single imp video request
+	for _, videoImp := range videoImps {
+		reqCopy.Imp = []openrtb.Imp{videoImp}
+		adapterReq, errors := a.makeRequest(&reqCopy)
+		if adapterReq != nil {
+			adapterRequests = append(adapterRequests, adapterReq)
+		}
+		errs = append(errs, errors...)
+	}
+
+	// Somoaudience only supports single imp video request
```

Comment:
another cut and paste comment mistake. :)

Reply:
This was a simple copy and paste mistake :)